qcodes/instrument/ip.py

```diff
@@ -142,10 +166,18 @@ def _send(self, cmd):
         self._socket.sendall(data.encode())
 
     def _recv(self):
-        result = self._socket.recv(self._buffer_size)
-        if result == b'':
-            log.warning("Got empty response from Socket recv() "
-                        "Connection broken.")
+        got_read_terminator = False
+        rtl = len(self.read_terminator)
+        result = b''
+        while not got_read_terminator:
+            partresult = self._socket.recv(self._buffer_size)
+            result += partresult
+            if partresult == b'':
+                log.warning("Got empty response from Socket recv() "
+                            "Connection broken.")
+            elif (partresult[-rtl:] == self.read_terminator or
```

Comment:
I would use "endswith" here: 
https://www.tutorialspoint.com/python/string_endswith.htm

Reply:
Agreed that is slightly more clean